.circleci/config.yml

```diff
@@ -211,54 +211,42 @@ jobs:
       - run:
           name: Run test vectors
           command: make run-vectors
-  # TODO: change executor to ubuntu
   coverage:
-    executor: mac-executor
+    executor: test-executor
     description: Run coverage reporting using grcov
     steps:
-      - checkout
+      - build_setup
+      - restore_cargo_package_cache
       - when:
           condition:
             equal: [main, << pipeline.git.branch >>]
           steps:
-          - run:
-              name: Install rust toolchain
-              command: |
-                curl https://sh.rustup.rs -sSf -o install_rust.sh
-                chmod +x install_rust.sh
-                ./install_rust.sh -q -y
-                rm install_rust.sh
-                export PATH="${HOME}/.cargo/bin:${PATH}"
-                rustc -V
-          - env_setup
           - run:
               name: Set rustc version
               command: |
                 rustup default nightly
                 rustup update nightly
-          - run:
-              name: Install deps
-              command: |
-                brew install hwloc
           - run:
               name: Install grcov
               command: |
-                cargo install grcov
+                cargo install --force grcov
           - run:
               name: Avoid hosts unknown for github
               command: echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config
           - run:
               name: Run test suite
               command: |
-                export RUSTFLAGS="-Zprofile -Ccodegen-units=1 -Coverflow-checks=off"
-                make build
-                git submodule update --init
+                export RUSTFLAGS="-g -Zprofile -Ccodegen-units=1 -Coverflow-checks=off"
+                export CARGO_INCREMENTAL=0
+                cargo clean
+                cargo build --release
                 sudo make test-all
           - run:
               name: Upload to codecov
               command: |
-                grcov ./target/debug -s . -t lcov --llvm --branch --ignore-not-existing --ignore="tests/*" --ignore="target/debug/build/*" --ignore="**/tests/*" -o lcov.info;
+                grcov ./target/release -s . -t lcov --llvm --branch --ignore-not-existing --ignore="tests/*" --ignore="target/release/build/*" --ignore="**/tests/*" -o lcov.info;
                 bash <(curl -s https://codecov.io/bash) -f lcov.info;
+                rm -f lcov.info
```

Comment:
Curious if this will fix the incorrect coverage report % in codecov. I know locally what was working for me was using:

`-t coveralls -s . --token <TOKEN> > coveralls.json` then sending the json to codecov.